lib/streamlit/elements/widgets/multiselect.py

```diff
@@ -274,9 +373,15 @@ def _multiselect(
         maybe_raise_label_warnings(label, label_visibility)
 
         indexable_options = convert_to_sequence_and_check_comparable(options)
-        formatted_options = [format_func(option) for option in indexable_options]
+        formatted_option_to_option_mapping: dict[str, T] = {
+            format_func(option): option for option in indexable_options
+        }
+        formatted_options: list[str] = list(formatted_option_to_option_mapping.keys())
         default_values = get_default_indices(indexable_options, default)
 
+        if accept_new_options:
+            placeholder = "Choose or create an option"
+
```

Comment:
Same as the comment on selectbox, we might only want to set this if placeholder wasn't explicitly provided by the user.

Reply:
Replied [here](https://github.com/streamlit/streamlit/pull/10788#discussion_r2012671392)